### PR TITLE
Core transaction processing logic

### PR DESCRIPTION
**Pull request by dlubarov**
With lots of TODOs to fill in afterward; this is just a start.

Depends on #651.

### CODE REVIEW

File: evm/src/cpu/columns/mod.rs [hunk 1]
```diff
@@ -17,9 +17,6 @@ pub struct CpuColumnsView<T: Copy> {
     /// Filter. 1 if the row is part of bootstrapping the kernel code, 0 otherwise.
     pub is_bootstrap_kernel: T,
 
-    /// Filter. 1 if the row is part of bootstrapping a contract's code, 0 otherwise.
-    pub is_bootstrap_contract: T,
```
Comment (dlubarov) [inline evm/src/cpu/columns/mod.rs:21]: (No longer needed; we're going to bootstrap contracts in kernel assembly now.)

File: evm/src/cpu/kernel/asm/core/call.asm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+global staticcall:
+    // TODO: Set flag to indicate
+    %jump(call_common)
+
+global delegatecall:
+    // TODO:
+    %jump(call_common)
+
+call_common:
+    %create_context
```
Comment (nbgl) [inline evm/src/cpu/kernel/asm/core/call.asm:16]: I had a look at this and at `create_context` and this is indeed what I was imagining for contract calls.

File: evm/src/cpu/kernel/asm/core/terminate.asm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// - the instruction is invalid
+// - there are insufficient stack items
+// - a JUMP/JUMPI destination is invalid
+// - the new stack size would be larger than 1024, or
+// - state modification is attempted during a static call
+global exception:
+    // TODO
+    %jump(terminate_common)
+
+terminate_common:
```
Comment (nbgl) [inline evm/src/cpu/kernel/asm/core/terminate.asm:34]: Do we want to charge gas here or has that been done?

Comment (dlubarov) [reply evm/src/cpu/kernel/asm/core/terminate.asm:34]: Like charge the cost of these terminal instructions? It looks like `STOP`, `RETURN` and `REVERT` are free (in fact the only free ones). The specs do have a "zero gas" constant for them, so we could do `%consume_gas_const(@GAS_ZERO)` and optimize it out if we wanted to match.

Or charge gas for the transaction? I believe that's done at the start, and then leftover gas is refunded upon graceful termination (just not exceptions). I'll add a TODO to refund gas to the parent context upon `STOP`, `RETURN` and `REVERT` or `SELFDESTRUCT`.

**Pull request merged**